Update/Remove tests relying on 'getPropertyCSSValue'

Ahmad-S792 opened the pull request:
#### 1e3966d2e89c2d1356ee89cd1de68c2daa150f63
<pre>
Update/Remove tests relying on &apos;getPropertyCSSValue&apos;

<a href="https://bugs.webkit.org/show_bug.cgi?id=229912">https://bugs.webkit.org/show_bug.cgi?id=229912</a>

Reviewed by Tim Nguyen.

This PR is to update tests relying on historical &apos;CSS&apos; properties and make it easier to remove them in future.

Selective Merge: <a href="https://chromium.googlesource.com/chromium/blink/+/018db695b2fd41eccdf23a87f157611f0425465d">https://chromium.googlesource.com/chromium/blink/+/018db695b2fd41eccdf23a87f157611f0425465d</a> ,
<a href="https://chromium.googlesource.com/chromium/blink/+/088bcb0bb1fc92f4144303a6650fcdccaacac5f9">https://chromium.googlesource.com/chromium/blink/+/088bcb0bb1fc92f4144303a6650fcdccaacac5f9</a> ,
<a href="https://chromium.googlesource.com/chromium/blink/+/16138b933dbebe3e3570464c8e3cf01d530f2bcd">https://chromium.googlesource.com/chromium/blink/+/16138b933dbebe3e3570464c8e3cf01d530f2bcd</a> ,
<a href="https://chromium.googlesource.com/chromium/blink/+/839bdc83ee530f8b972f4195587995a7bda22368">https://chromium.googlesource.com/chromium/blink/+/839bdc83ee530f8b972f4195587995a7bda22368</a> ,
<a href="https://chromium.googlesource.com/chromium/blink/+/47b9403fd8b928a324d35a71566548ce6592568a">https://chromium.googlesource.com/chromium/blink/+/47b9403fd8b928a324d35a71566548ce6592568a</a> ,
<a href="https://chromium.googlesource.com/chromium/blink/+/98d37e56283bb920a7ec652ea9235f8f5fd87a7a">https://chromium.googlesource.com/chromium/blink/+/98d37e56283bb920a7ec652ea9235f8f5fd87a7a</a> ,
<a href="https://chromium.googlesource.com/chromium/blink/+/ab1811d8bf4e7fa18f9c6fa44e5a86bb7b2f7d17">https://chromium.googlesource.com/chromium/blink/+/ab1811d8bf4e7fa18f9c6fa44e5a86bb7b2f7d17</a> ,
<a href="https://chromium.googlesource.com/chromium/blink/+/391870c98875dcff621ab992553ebf9f4f695012">https://chromium.googlesource.com/chromium/blink/+/391870c98875dcff621ab992553ebf9f4f695012</a> ,
<a href="https://chromium.googlesource.com/chromium/blink/+/42fab77317a1ba7b9b91225c64460fff932d6312">https://chromium.googlesource.com/chromium/blink/+/42fab77317a1ba7b9b91225c64460fff932d6312</a> ,
<a href="https://chromium.googlesource.com/chromium/blink/+/d264cf786c1b2a87f99e095e09ea00bb85ee4e69">https://chromium.googlesource.com/chromium/blink/+/d264cf786c1b2a87f99e095e09ea00bb85ee4e69</a> ,
<a href="https://chromium.googlesource.com/chromium/blink/+/2c7a5a8bf7d3f37e0af669bb241886e3aceffbfb">https://chromium.googlesource.com/chromium/blink/+/2c7a5a8bf7d3f37e0af669bb241886e3aceffbfb</a> and
<a href="https://chromium.googlesource.com/chromium/blink/+/b179bc4d8475adc7943745c6cabe6997202577dc">https://chromium.googlesource.com/chromium/blink/+/b179bc4d8475adc7943745c6cabe6997202577dc</a>

&gt; Rebaselined Tests &amp; Expectations:
* LayoutTests/animations/fill-mode-iteration-count-non-integer.html:
* LayoutTests/animations/fill-mode-missing-from-to-keyframes.html:
* LayoutTests/animations/fill-mode-multiple-keyframes.html:
* LayoutTests/animations/fill-mode-removed.html:
* LayoutTests/animations/fill-mode.html:
* LayoutTests/css3/blending/background-blend-mode-property-parsing.html:
* LayoutTests/css3/blending/background-blend-mode-property-parsing-expected.txt:
* LayoutTests/css3/blending/blend-mode-property-parsing.html:
* LayoutTests/css3/blending/blend-mode-property-parsing-expected.txt:
* LayoutTests/css3/calc/reflection-computed-style.html:
* LayoutTests/css3/calc/reflection-computed-style-expected.txt:
* LayoutTests/css3/filters/filter-property-parsing.html:
* LayoutTests/css3/filters/filter-property-parsing-expected.txt:
* LayoutTests/css3/filters/filter-property-computed-style.html:
* LayoutTests/css3/filters/filter-property-computed-style-expected.txt:
* LayoutTests/fast/css/custom-font-xheight.html:
* LayoutTests/fast/css/sheet-collection-link.html:
* LayoutTests/fast/css/style-enumerate-properties.html:
* LayoutTests/fast/css/style-enumerate-properties-expected.txt:
* LayoutTests/fast/css3-text/css3-text-decoration/getComputedStyle/getComputedStyle-text-decoration-color.html:
* LayoutTests/fast/css3-text/css3-text-decoration/getComputedStyle/getComputedStyle-text-decoration-color-expected.txt:
* LayoutTests/fast/css3-text/css3-text-decoration/getComputedStyle/getComputedStyle-text-decoration-line.html:
* LayoutTests/fast/css3-text/css3-text-decoration/getComputedStyle/getComputedStyle-text-decoration-line-expected.txt:
* LayoutTests/fast/css3-text/css3-text-decoration/getComputedStyle/getComputedStyle-text-decoration-style.html;
* LayoutTests/fast/css3-text/css3-text-decoration/getComputedStyle/getComputedStyle-text-decoration-style-expected.txt:
* LayoutTests/fast/css3-text/css3-text-decoration/getComputedStyle/getComputedStyle-text-underline-position.html:
* LayoutTests/fast/css3-text/css3-text-decoration/getComputedStyle/getComputedStyle-text-underline-position-expected.txt:
* LayoutTests/fast/dom/css-element-attribute-js-null.html:
* LayoutTests/fast/dom/css-element-attribute-js-null-expected.txt:
* LayoutTests/fast/dom/dom-constructors-expected.txt:
* LayoutTests/fast/dom/dom-constructors.html:
* LayoutTests/fast/dom/domListEnumeration.html:
* LayoutTests/fast/dom/domListEnumeration-expected.txt:
* LayoutTests/fast/dom/gc-9.html:
* LayoutTests/fast/dom/gc-9-expected.txt:
* LayoutTests/fast/dom/non-numeric-values-numeric-parameters.html:
* LayoutTests/fast/dom/non-numeric-values-numeric-parameters-expected.txt:
* LayoutTests/fast/dom/prototypes.html:
* LayoutTests/fast/dom/prototypes-expected.txt:
* LayoutTests/fast/dom/wrapper-classes.html:
* LayoutTests/fast/dom/wrapper-classes-expected.txt:
* LayoutTests/fast/history/visited-generated-content-test.html:
* LayoutTests/fast/media/mq-js-update-media.html:
* LayoutTests/fast/reflections/reflection-computed-style.html:
* LayoutTests/fast/reflections/reflection-computed-style-expected.txt:
* LayoutTests/platform/mac/fast/dom/objc-wrapper-identity.html: Just ‘js-test’ update
* LayoutTests/platform/mac/fast/dom/objc-wrapper-identity-expected.txt: Just ‘js-test’ update
* LayoutTests/svg/animations/animate-color-rgba-calcMode-discrete.html:
* LayoutTests/svg/animations/animate-color-rgba-calcMode-discrete-expected.txt:
* LayoutTests/svg/animations/change-css-property-while-animating-fill-freeze.html:
* LayoutTests/svg/animations/change-css-property-while-animating-fill-freeze-expected.txt:
* LayoutTests/svg/animations/change-css-property-while-animating-fill-remove.html:
* LayoutTests/svg/animations/change-css-property-while-animating-fill-remove-expected.txt:
* LayoutTests/svg/css/getComputedStyle-basic.xhtml:
* LayoutTests/svg/css/getComputedStyle-basic-expected.txt:
* LayoutTests/svg/css/glyph-orientation-rounding-test.xhtml:
* LayoutTests/svg/css/glyph-orientation-rounding-test-expected.txt:
* LayoutTests/svg/css/rect-system-color.xhtml:
* LayoutTests/svg/transforms/transform-origin-presentation-attribute.xhtml:

&gt; Deleted Tests:
* LayoutTests/http/tests/security/contentSecurityPolicy/inline-style-allowed-while-cloning-objects.html: Deleted in favour of WPT Test
* LayoutTests/http/tests/security/contentSecurityPolicy/inline-style-allowed-while-cloning-objects-expected.txt: Ditto

Canonical link: <a href="https://commits.webkit.org/272756@main">https://commits.webkit.org/272756@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/081cfdc482f48bfc1d529a26ec0bd09040817faf

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/32943 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/11703 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/34826 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/35571 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/29761 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/14044 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/8879 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/29205 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/33403 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/9860 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/29425 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/8600 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/8742 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/29381 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/36903 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/29917 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/29791 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/34881 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/8874 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/6827 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/32736 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/10568 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/29038 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/7644 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/9478 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/9503 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m reviewed (commented):
I'm OK changing existing tests to not rely on `getPropertyCSSValue`. This seems like generally improvements.

However, we shouldn't remove:
- tests that cover very basic `getPropertyCSSValue` functionality
- tests that cover security / memory issues (GC / crash tests)

nt1m commented on LayoutTests/http/tests/security/cross-frame-access-get.html:
```diff
@@ -7,10 +7,8 @@
             "CSSRule", 
             "CSSStyleDeclaration", 
-            "CSSValue", 
             "CharacterData", 
```
I would restore these, and only remove these when the functionality has been removed.

nt1m commented on LayoutTests/http/tests/security/cross-frame-access-getOwnPropertyDescriptor.html:
```diff
@@ -7,10 +7,8 @@
-            "CSSPrimitiveValue", 
             "CSSRule", 
             "CSSStyleDeclaration", 
-            "CSSValue", 
```
same here

nt1m commented on LayoutTests/http/tests/security/cross-frame-access-put.html:
```diff
@@ -34,10 +34,8 @@
-    setForbiddenProperty(targetWindow, "CSSPrimitiveValue");
     setForbiddenProperty(targetWindow, "CSSRule");
     setForbiddenProperty(targetWindow, "CSSStyleDeclaration");
-    setForbiddenProperty(targetWindow, "CSSValue");
```
same here

nt1m commented on LayoutTests/http/tests/security/resources/cross-frame-iframe-for-put-test.html:
Same throughout this file

Ahmad-S792 replied:
Done.

nt1m commented on LayoutTests/fast/dom/Window/resources/window-properties.js:
```diff
@@ -123,7 +119,6 @@ var propertyInfo = [
     ["ProcessingInstruction", "object"],
     ["ProgressEvent", "object"],
     ["Range", "object"],
-    ["Rect", "object"],
```
I would also restore all of these, until the functionality actually gets removed.

nt1m commented on LayoutTests/fast/dom/Window/get-set-properties.html:
```diff
@@ -74,7 +74,6 @@
     "CSSPrimitiveValue", 
     "CSSRule", 
     "CSSStyleDeclaration", 
-    "CSSValue", 
```
I would also keep this.

Ahmad-S792 replied:
Done.

nt1m reviewed (approved):
Most of the patch looks reasonable! Thanks for the cleanup

Ahmad-S792 committed:
Update/Remove tests relying on 'getPropertyCSSValue'

https://bugs.webkit.org/show_bug.cgi?id=229912

Reviewed by Tim Nguyen.

This PR is to update tests relying on historical 'CSS' properties and make it easier to remove them in future.

Selective Merge: https://chromium.googlesource.com/chromium/blink/+/018db695b2fd41eccdf23a87f157611f0425465d ,
https://chromium.googlesource.com/chromium/blink/+/088bcb0bb1fc92f4144303a6650fcdccaacac5f9 ,
https://chromium.googlesource.com/chromium/blink/+/16138b933dbebe3e3570464c8e3cf01d530f2bcd ,
https://chromium.googlesource.com/chromium/blink/+/839bdc83ee530f8b972f4195587995a7bda22368 ,
https://chromium.googlesource.com/chromium/blink/+/47b9403fd8b928a324d35a71566548ce6592568a ,
https://chromium.googlesource.com/chromium/blink/+/98d37e56283bb920a7ec652ea9235f8f5fd87a7a ,
https://chromium.googlesource.com/chromium/blink/+/ab1811d8bf4e7fa18f9c6fa44e5a86bb7b2f7d17 ,
https://chromium.googlesource.com/chromium/blink/+/391870c98875dcff621ab992553ebf9f4f695012 ,
https://chromium.googlesource.com/chromium/blink/+/42fab77317a1ba7b9b91225c64460fff932d6312 ,
https://chromium.googlesource.com/chromium/blink/+/d264cf786c1b2a87f99e095e09ea00bb85ee4e69 ,
https://chromium.googlesource.com/chromium/blink/+/2c7a5a8bf7d3f37e0af669bb241886e3aceffbfb and
https://chromium.googlesource.com/chromium/blink/+/b179bc4d8475adc7943745c6cabe6997202577dc

> Rebaselined Tests & Expectations:
* LayoutTests/animations/fill-mode-iteration-count-non-integer.html:
* LayoutTests/animations/fill-mode-missing-from-to-keyframes.html:
* LayoutTests/animations/fill-mode-multiple-keyframes.html:
* LayoutTests/animations/fill-mode-removed.html:
* LayoutTests/animations/fill-mode.html:
* LayoutTests/css3/blending/background-blend-mode-property-parsing.html:
* LayoutTests/css3/blending/background-blend-mode-property-parsing-expected.txt:
* LayoutTests/css3/blending/blend-mode-property-parsing.html:
* LayoutTests/css3/blending/blend-mode-property-parsing-expected.txt:
* LayoutTests/css3/calc/reflection-computed-style.html:
* LayoutTests/css3/calc/reflection-computed-style-expected.txt:
* LayoutTests/css3/filters/filter-property-parsing.html:
* LayoutTests/css3/filters/filter-property-parsing-expected.txt:
* LayoutTests/css3/filters/filter-property-computed-style.html:
* LayoutTests/css3/filters/filter-property-computed-style-expected.txt:
* LayoutTests/fast/css/custom-font-xheight.html:
* LayoutTests/fast/css/sheet-collection-link.html:
* LayoutTests/fast/css/style-enumerate-properties.html:
* LayoutTests/fast/css/style-enumerate-properties-expected.txt:
* LayoutTests/fast/css3-text/css3-text-decoration/getComputedStyle/getComputedStyle-text-decoration-color.html:
* LayoutTests/fast/css3-text/css3-text-decoration/getComputedStyle/getComputedStyle-text-decoration-color-expected.txt:
* LayoutTests/fast/css3-text/css3-text-decoration/getComputedStyle/getComputedStyle-text-decoration-line.html:
* LayoutTests/fast/css3-text/css3-text-decoration/getComputedStyle/getComputedStyle-text-decoration-line-expected.txt:
* LayoutTests/fast/css3-text/css3-text-decoration/getComputedStyle/getComputedStyle-text-decoration-style.html;
* LayoutTests/fast/css3-text/css3-text-decoration/getComputedStyle/getComputedStyle-text-decoration-style-expected.txt:
* LayoutTests/fast/css3-text/css3-text-decoration/getComputedStyle/getComputedStyle-text-underline-position.html:
* LayoutTests/fast/css3-text/css3-text-decoration/getComputedStyle/getComputedStyle-text-underline-position-expected.txt:
* LayoutTests/fast/dom/css-element-attribute-js-null.html:
* LayoutTests/fast/dom/css-element-attribute-js-null-expected.txt:
* LayoutTests/fast/dom/dom-constructors-expected.txt:
* LayoutTests/fast/dom/dom-constructors.html:
* LayoutTests/fast/dom/domListEnumeration.html:
* LayoutTests/fast/dom/domListEnumeration-expected.txt:
* LayoutTests/fast/dom/gc-9.html:
* LayoutTests/fast/dom/gc-9-expected.txt:
* LayoutTests/fast/dom/non-numeric-values-numeric-parameters.html:
* LayoutTests/fast/dom/non-numeric-values-numeric-parameters-expected.txt:
* LayoutTests/fast/dom/prototypes.html:
* LayoutTests/fast/dom/prototypes-expected.txt:
* LayoutTests/fast/dom/wrapper-classes.html:
* LayoutTests/fast/dom/wrapper-classes-expected.txt:
* LayoutTests/fast/history/visited-generated-content-test.html:
* LayoutTests/fast/media/mq-js-update-media.html:
* LayoutTests/fast/reflections/reflection-computed-style.html:
* LayoutTests/fast/reflections/reflection-computed-style-expected.txt:
* LayoutTests/platform/mac/fast/dom/objc-wrapper-identity.html: Just ‘js-test’ update
* LayoutTests/platform/mac/fast/dom/objc-wrapper-identity-expected.txt: Just ‘js-test’ update
* LayoutTests/svg/animations/animate-color-rgba-calcMode-discrete.html:
* LayoutTests/svg/animations/animate-color-rgba-calcMode-discrete-expected.txt:
* LayoutTests/svg/animations/change-css-property-while-animating-fill-freeze.html:
* LayoutTests/svg/animations/change-css-property-while-animating-fill-freeze-expected.txt:
* LayoutTests/svg/animations/change-css-property-while-animating-fill-remove.html:
* LayoutTests/svg/animations/change-css-property-while-animating-fill-remove-expected.txt:
* LayoutTests/svg/css/getComputedStyle-basic.xhtml:
* LayoutTests/svg/css/getComputedStyle-basic-expected.txt:
* LayoutTests/svg/css/glyph-orientation-rounding-test.xhtml:
* LayoutTests/svg/css/glyph-orientation-rounding-test-expected.txt:
* LayoutTests/svg/css/rect-system-color.xhtml:
* LayoutTests/svg/transforms/transform-origin-presentation-attribute.xhtml:

> Deleted Tests:
* LayoutTests/http/tests/security/contentSecurityPolicy/inline-style-allowed-while-cloning-objects.html: Deleted in favour of WPT Test
* LayoutTests/http/tests/security/contentSecurityPolicy/inline-style-allowed-while-cloning-objects-expected.txt: Ditto

Canonical link: https://commits.webkit.org/272756@main
